app/storage/document-repo.js

```diff
@@ -0,0 +1,42 @@
+const { v4: uuidv4 } = require('uuid')
+
+const { blobServiceClient } = require('./blob-service-client')
+const config = require('../config/storage')
+
+const saveDocument = async (buffer, type) => {
+  const documentsContainer = blobServiceClient.getContainerClient(config.documentsContainer)
+
+  const id = uuidv4()
+
+  const blockBlobClient = documentsContainer.getBlockBlobClient(id)
+
+  const options = {
+    blobHTTPHeaders: {
+      blobContentType: type
+    }
+  }
+
+  await blockBlobClient.uploadData(buffer, options)
+
+  return id
+}
+
+const updateDocumentMetadata = async (id, metadata) => {
+  const documentsContainer = blobServiceClient.getContainerClient(config.documentsContainer)
+  const blockBlobClient = documentsContainer.getBlockBlobClient(id)
+
+  if (!await blockBlobClient.exists()) {
+    const err = new Error(`The document with ID ${id} does not exist`)
+
+    err.code = 'NotFound'
+
+    throw err
+  }
+
+  await blockBlobClient.setMetadata(metadata)
```

Comment:
Blob storage stores metadata keys as lower case regardless of the input. Thinking of the GET endpoints incoming, it will probably be best to have these in snake case for readability.

See [coreai-mcu-knowledge-pgv/app/mappers/knowledge-metadata.js](https://github.com/DEFRA/coreai-mcu-knowledge-pgv/blob/2fd21219b063b0766ddf95eee8a82e9c7cf96142/app/mappers/knowledge-metadata.js)

Reply:
changes made to handle this.